src/ORM/DataObjectSchema.php

```diff
@@ -471,7 +479,13 @@ protected function cacheDatabaseFields($class)
             $fieldObj->setTable($class);
             $nestedFields = $fieldObj->compositeDatabaseFields();
             foreach ($nestedFields as $nestedName => $nestedSpec) {
-                $dbFields["{$fieldName}{$nestedName}"] = $nestedSpec;
+                $compositeName = $fieldName . $nestedName;
+                $dbFields[$compositeName] = $nestedSpec;
+
+                // Handle configurable indexes, e.g. for polymorphic keys
```

Comment:
My suggestion for using fieldSpecs should pull those out in the result. Although, you'll get both the composite field parent, and all sub-fields in the result. Not an issue unless a composite field has 'indexed' and one of it's components ALSO has 'indexed' though right?